src/Internal/Payment/State/StateFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Class StateFactory
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WCPay\Container;
+use WCPay\Vendor\League\Container\Exception\ContainerException;
+use WCPay\Internal\Payment\Exception\StateTransitionException;
+use WCPay\Internal\Payment\PaymentContext;
+
+/**
+ * A factory for payment states.
+ *
+ * The main purpose of this class is to work as an extension of the
+ * WooPayments dependency container, only allowing access to payment states
+ */
+class StateFactory {
+	/**
+	 * Holds the WooPayments DI container.
+	 *
+	 * @var Container
+	 */
+	private $container;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param Container $container Dependency container.
+	 */
+	public function __construct( Container $container ) {
+		$this->container = $container;
+	}
+
+	/**
+	 * Creates a new state based on class name.
+	 *
+	 * @param string         $state_class Name of the state class.
+	 * @param PaymentContext $context     Context for the new state.
+	 * @return PaymentState               The generated payment state instance.
+	 * @throws StateTransitionException   When the class name is not a state.
+	 */
+	public function create_state( string $state_class, PaymentContext $context ): PaymentState {
```

Comment:
I believe we can eliminate this class and transfer its logic to the `PaymentContext` class. By enforcing that state classes implement a specific interface or contract, we can force certain class behaviors if needed.